101-rds-deployment-full-ha/azuredeploy.json

```diff
@@ -0,0 +1,1180 @@
+                            "protocol": "Tcp",
+                            "direction": "Inbound",
+                            "priority": 1001                            
+                        }
+                    },
+                    {
+                        "name": "Allow_HTTPS",
+                        "properties": {
+                            "access": "Allow",
+                            "sourceAddressPrefix": "*",
+                            "sourcePortRange": "*",
+                            "destinationAddressPrefix": "*",
+                            "destinationPortRange": "443",
+                            "protocol": "Tcp",
+                            "direction": "Inbound",
+                            "priority": 1002                        
+                        }
+                    },
+                    {
+                        "name": "Allow_UDP_3391",
+                        "properties": {
+                            "access": "Allow",
+                            "sourceAddressPrefix": "*",
+                            "sourcePortRange": "*",
+                            "destinationAddressPrefix": "*",
+                            "destinationPortRange": "3391",
+                            "protocol": "Udp",
+                            "direction": "Inbound",
+                            "priority": 1003                        
+                        }
+                    }                                     
+                ]
+            }
+        },
+        {
+            "type": "Microsoft.Network/virtualNetworks",
+            "apiVersion": "2019-12-01",
+            "location": "[parameters('location')]",
+            "name": "[concat(parameters('projectName'),'vnet')]",
+            "dependsOn": [
+                "[variables('nsgRef')]"
+            ],
+            "properties": {
+                "addressSpace": {
+                    "addressPrefixes": [
+                        "[parameters('vNetAddressSpace')]"
+                    ]
+                },
+                "subnets": [
+                    {
+                        "name": "[concat(parameters('projectName'),'main')]",
+                        "properties": {
+                            "addressPrefix": "[parameters('vNetSubnetAddress')]",
+                            "networkSecurityGroup": {
+                                "id": "[variables('nsgRef')]"
+                            }
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "type": "Microsoft.Network/loadBalancers",
+            "apiVersion": "2019-12-01",
+            "location": "[parameters('location')]",
+            "name": "[concat(parameters('projectName'),'intlb')]",
+            "sku": {
+                "name": "Standard"
+            },
+            "dependsOn": [
+                "[concat(parameters('projectName'),'vnet')]"

```

Comment:
remove concat and use resourceId, ie.

[resourceId('Microsoft.Network/virtualNetworks', concat(parameters('projectName'),'vnet'))]